src/lib/reasoners/satml_frontend.ml

```diff
@@ -1116,9 +1104,10 @@ module Make (Th : Theory.S) : Sat_solver_sig.S = struct
         (* The answer for the first objective is infinity. We stop here as
            we cannot go beyond infinity and the next objectives with lower
            priority cannot be optimized in presence of infinity values. *)
-        raise unknown_exn;
+        raise I_dont_know;
 
       | (e, tv, is_max, is_le) :: l ->
+        env.objectives <- Some obj;
```

Comment:
Can you explain why this is moved here and why it is correct? Is it related to the test failure you observed?

Reply:
In the old version, if `acc` is empty, the objective update will not be taken into account because an `I_dont_know` exception is raised. It only is otherwise that the objectives must be updated.

Reply:
Oh you're right I forgot it was a reference; and no, it is not what fixes the test failure.

Reply:
Actually the goal of this reference isn't to be able retrieve objectives (currently, objectives are stored in a struct from `Models.ml` and the reference `last_saved_model` gives us access to them) but to keep in mind we're exploring branches beyond SAT because we're looking for a better model during the optimization. If we got an `unsat` answer  in this context, it means the last attempt to improve the model failed but the previous one is correct and we have to return `unknown` for the frontend. The critical code is here:
 ```ocaml
      begin
        try analyze_unknown_for_objectives env e unsat_rec_prem
        with
        | IUnsat (env, _) ->
          assert (!(env.objectives) != None);
          (* objectives is a ref, it's necessiraly updated as a
             side-effect to best value *)
          raise (I_dont_know env)
      end
```